packages/core/src/layout/InfoCard/InfoCard.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { FC, ReactNode } from 'react';
+import {
+  Card,
+  CardActions,
+  CardContent,
+  CardHeader,
+  Divider,
+  withStyles,
+  makeStyles,
+} from '@material-ui/core';
+import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
+import BottomLink from './BottomLink';
+import { BackstageTheme } from '../../theme/theme';
+
+const useStyles = makeStyles<BackstageTheme>(theme => ({
+  header: {
+    padding: theme.spacing(2, 2, 2, 2.5),
+  },
+}));
+
+const BoldHeader = withStyles({
+  title: { fontWeight: 700 },
+  subheader: { paddingTop: '2px' },
+})(CardHeader);
+
+const CardActionsTopRight = withStyles({
+  root: {
+    display: 'inline-block',
+    paddingRight: '16px',
+    paddingTop: '16px',
+    float: 'right',
+  },
+})(CardActions);
+
+const VARIANT_STYLES = {
+  card: {
+    flex: {
```

Comment:
all of these is such an atrocity, there's gotta be a better way ...

(@hooloovooo this is just me whining about our legacy, not meant as an actionable comment :P)

Reply:
Agreed. But if they are in more places it might be a good idea to figure out something better and then replace all of them. :)

Reply:
Yeah. I'm thinking it's better that the caller can explicitly do `<InfoCard className={classes.myFullHeightCss}>` or using the `classes` mechanism or whatever makes the most sense and is the most material-ui-embraced way of doing child styling. I see some MUI things use props with an uppercase name like `<SomeComponent SubComponentProps={{ ... }}>` or whatever; that way we could expose direct styling of the underlying card while still being able to benefit from the use of consolidated/standardising wrappers

Reply:
The variants also contain some magic pixel values (header height etc) that i think would be nice to define somewhere else (theme?).

Reply:
They are confusing to me in general. That kind of fiddling should never be necessary unless you are like absolutely positioning something on top of other stuff or whatever. Should be handled instead by the regular DOM layout making space for whatever you put inside blocks.

Reply:
Is this something we want to fix in this PR, or should we create a new task for cleaning up this?

Reply:
That should be a separate task, for sure. You've already done great here.